### ISSUE TITLE
Cannot Download any HP Driver FTP 

### ISSUE THREAD

**Issue by Mrgreen17** [maurice-daly/DriverAutomationTool#454]
i can manually download the driver packs but it is not working thru the tool. 

here is just the error:

**[Error] - An error occurred while attempting contact http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/89d2/89d2_64_22H2.cab - The remote server returned an error: (404) Not Found.	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)**
 	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)
- Could not download HP EliteBook 645 14 inch G9 Notebook PC BIOS update	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)

here is the full log 

======== Processing HP EliteBook 645 14 inch G9 Notebook PC Downloads ========	DriverAutomationTool	1/27/2023 9:37:29 AM	9692 (0x25DC)
- Starting Download, extract and import processes for HP model: EliteBook 645 14 inch G9 Notebook PC	DriverAutomationTool	1/27/2023 9:37:29 AM	9692 (0x25DC)
- Setting HP variables	DriverAutomationTool	1/27/2023 9:37:29 AM	9692 (0x25DC)
- Using Configuration Manager console BIOS package folder - SH1:\Package\BIOS Packages\HP	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- Using Configuration Manager console Driver package folder - SH1:\Package\Driver Packages\HP	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
======== HP EliteBook 645 14 inch G9 Notebook PC BIOS PROCESSING STARTED ========	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- Attempting to find HP BIOS download	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- Checking for existing HP cabinet file platformList.cab	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- Reading HP XML from C:\Program Files\MSEndpointMgr\Driver Automation Tool\Temp\platformList.xml	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- OS pre build strip is 22H2	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- Model is EliteBook 645 14 inch G9 Notebook PC	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- SystemID is 89d2	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- OS is 22H2	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- Architecture is x64	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
- URL is http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/89d2/89d2_64_22H2.cab	DriverAutomationTool	1/27/2023 9:37:38 AM	9692 (0x25DC)
======== Errors(s) Occurred ========	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)
**[Error] - An error occurred while attempting contact http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/89d2/89d2_64_22H2.cab - The remote server returned an error: (404) Not Found.	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)**
 	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)
- Could not download HP EliteBook 645 14 inch G9 Notebook PC BIOS update	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)
Warning: Unable to retrieve BIOS Download URL For HP Client Model: EliteBook 645 14 inch G9 Notebook PC	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)
======== HP EliteBook 645 14 inch G9 Notebook PC BIOS PROCESSING FINISHED ========	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)
======== Superseded BIOS Package Option Processing ========	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)
- Remaining models to process: 0	DriverAutomationTool	1/27/2023 9:37:39 AM	9692 (0x25DC)

**Comment by XchoK** [comment]
@Mrgreen17 I am having the same issue with HP models for the BIOS. The drivers work. Were you able to resolve this?

**Comment by stephannn** [comment]
it seems like HP changed the file URI from
`http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/89d2/89d2_64_22H2.cab`
to
`https://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/89d2/89d2_64_10.0.22h2.cab`

the OS version was added and the H in H2 is case sensitive.

If I understand the code correctly the issue is that , same as in the Invoke Driver Script, the function _Find-HPBIOS_ does not separate between Windows 10 and 11. Any double checking would be appreciated.

**Comment by Mrgreen17** [comment]
@xchoK you are correct the drivers are working but the bios for hp models are still getting the error. I have not resolved this.

**Comment by XchoK** [comment]
@stephannn Where did you change the http to https? Thanks!

**Comment by stephannn** [comment]
@XchoK ,
I did this change here:
`$BIOSDownload = "https://" + $($HPBIOSDownload.URL)`
right below
`# ================= HP BIOS Upgrade Download ==================`

I was just too lazy to put a global variable for http or https. I also published my changes here:
https://github.com/maurice-daly/DriverAutomationTool/pull/474

**Comment by korpx** [comment]
**I confirm this is still a problem
When will this be fixed in the Master branch?**

The tool gives the invalid URL;
https://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b50/8b50_64_22H2.cab

While this appear to be the correct URL;
https://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b50/8b50_64_10.0.22h2.cab



Full log for HP Pro x360 435 13.3 inch G10 Notebook PC BIOS

======== HP Pro x360 435 13.3 inch G10 Notebook PC BIOS PROCESSING STARTED ========	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
- Attempting to find HP BIOS download	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
- Checking for existing HP cabinet file platformList.cab	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
- Reading HP XML from C:\Program Files\MSEndpointMgr\Driver Automation Tool\Temp\platformList.xml	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
- OS pre build strip is 22H2	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
- Model is Pro x360 435 13.3 inch G10 Notebook PC	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
- SystemID is 8b50	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
- OS is 22H2	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
- Architecture is x64	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
- URL is http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b50/8b50_64_22H2.cab	DriverAutomationTool	2024-05-31 09:55:04	16504 (0x4078)
======== Error(s) Occurred ========	DriverAutomationTool	2024-05-31 09:55:05	16504 (0x4078)
[Error] - An error occurred while attempting contact http://ftp.hp.com/pub/caps-softpaq/cmit/imagepal/ref/8b50/8b50_64_22H2.cab - The remote server returned an error: (404) Not Found.	DriverAutomationTool	2024-05-31 09:55:05	16504 (0x4078)
 	DriverAutomationTool	2024-05-31 09:55:05	16504 (0x4078)
- Could not download HP Pro x360 435 13.3 inch G10 Notebook PC BIOS update	DriverAutomationTool	2024-05-31 09:55:05	16504 (0x4078)
Warning: Unable to retrieve BIOS Download URL For HP Client Model: Pro x360 435 13.3 inch G10 Notebook PC	DriverAutomationTool	2024-05-31 09:55:05	16504 (0x4078)
======== HP Pro x360 435 13.3 inch G10 Notebook PC BIOS PROCESSING FINISHED ========	DriverAutomationTool	2024-05-31 09:55:05	16504 (0x4078)
- Remaining models to process: 0	DriverAutomationTool	2024-05-31 09:55:05	16504 (0x4078)
======== Clean Up Driver Option Processing ========	DriverAutomationTool	2024-05-31 09:55:06	16504 (0x4078)
- ConfigMgr: Removing driver download and extracted source driver files from \\sdsrv002\Source$\OS\DATPkgDownload	DriverAutomationTool	2024-05-31 09:55:06	16504 (0x4078)
======== PROCESS SUMMARY ========	DriverAutomationTool	2024-05-31 09:55:06	16504 (0x4078)
======== FINISHED PROCESSING ========	DriverAutomationTool	2024-05-31 09:55:06	16504 (0x4078)
